pilot/pkg/config/kube/ingress/conversion.go

```diff
@@ -131,40 +159,37 @@ func ConvertIngressV1alpha3(ingress v1beta1.Ingress, domainSuffix string) (model
 			httpRoute.Match = []*networking.HTTPMatchRequest{httpMatch}
 			httpRoutes = append(httpRoutes, httpRoute)
 		}
-	}
-
-	if ingress.Spec.Backend != nil {
-		httpRoutes = append(httpRoutes, ingressBackendToHTTPRoute(ingress.Spec.Backend, ingress.Namespace, domainSuffix))
-	}
 
-	virtualService.Http = httpRoutes
+		virtualService.Http = httpRoutes
 
-	gatewayConfig := model.Config{
-		ConfigMeta: model.ConfigMeta{
-			Type:      model.Gateway.Type,
-			Group:     model.Gateway.Group,
-			Version:   model.Gateway.Version,
-			Name:      ingress.Name + "-" + model.IstioIngressGatewayName,
-			Namespace: model.IstioIngressNamespace,
-			Domain:    domainSuffix,
-		},
-		Spec: gateway,
-	}
+		virtualServiceConfig := model.Config{
+			ConfigMeta: model.ConfigMeta{
+				Type:      model.VirtualService.Type,
+				Group:     model.VirtualService.Group,
+				Version:   model.VirtualService.Version,
+				Name:      namePrefix + "-" + ingress.Name + "-" + model.IstioIngressGatewayName,
+				Namespace: model.IstioIngressNamespace,
+				Domain:    domainSuffix,
+			},
+			Spec: virtualService,
+		}
 
-	virtualServiceConfig := model.Config{
-		ConfigMeta: model.ConfigMeta{
-			Type:      model.VirtualService.Type,
-			Group:     model.VirtualService.Group,
-			Version:   model.VirtualService.Version,
-			Name:      ingress.Name + "-" + model.IstioIngressGatewayName,
-			Namespace: model.IstioIngressNamespace,
-			Domain:    domainSuffix,
-		},
-		Spec: virtualService,
+		old, f := ingressByHost[host]
+		if f {
+			vs := old.Spec.(*networking.VirtualService)
+			vs.Http = append(vs.Http, httpRoutes...)
+		} else {
+			ingressByHost[host] = &virtualServiceConfig
+		}
 	}
 
-	return gatewayConfig, virtualServiceConfig
+	// Matches * and "/". Currently not supported - would conflict
+	// with any other explicit VirtualService.
+	if ingress.Spec.Backend != nil {
+		log.Infof("Ignore default wildcard ingress, use VirtualService %s:%s",
+			ingress.Namespace, ingress.Name)
 
+	}
```

Comment:
looking over this function, i don't see anything that would require ingress to be modified in place. per earlier comment, i'd recommend returning a new object instead.